vars/rawBuild.groovy

```diff
@@ -0,0 +1,41 @@
+/**
+ * Checks whether the current build has an upstream cause
+ **/
+@NonCPS
+def hasUpstream() {
+    return currentBuild.rawBuild.getCause(hudson.model.Cause$UpstreamCause.class) != null
+}
+
+/**
+ * Navigates the upstream cause of the current cuild
+ **/
+@NonCPS
+def upstreamCause() {
+    return currentBuild.rawBuild.getCause(hudson.model.Cause$UpstreamCause.class)
+}
+
+/**
+ * Navigates the upstream run of the current cuild
+ **/
+@NonCPS
+def upstreamRun() {
+    def upstreamCause = upstreamCause()
+    if (upstreamCause == null) {
+        return null
+    }
+    return upstreamCause.upstreamRun
+}
```

Comment:
isn't it overkill? this looks like equivalent to Groovy Elvis operator: `upstreamCause?.upstreamRun`